includes/internal/emails/generators/class-email-generators-abstract.php

```diff
@@ -69,10 +69,23 @@ abstract public function init();
 	 * @return boolean Indicates if the email is published or not
 	 */
 	public function is_email_active() {
-		$email = $this->repository->get( static::IDENTIFIER_NAME );
+		$email = $this->repository->get( $this->get_identifier() );
 		return $email && 'publish' === $email->post_status;
 	}
 
+	/**
+	 * Get name of the identifier of the Email.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @return string Identifier name.
+	 */
+	public function get_identifier() {
```

Comment:
By the way, this solution is better not only from complex cases' perspective, but also in testing – we can't substitute the value of a constant without subclassing, but we can substitute the returning value of a public method in the mock.